pysph/tools/ipy_viewer.py

```diff
@@ -1189,8 +1459,76 @@ def _save_all_plots_handler(self, change):
 
             for array_name in self._widgets.particles.keys():
                 pa_widgets = self._widgets.particles[array_name]
-                pa_widgets.scalar.disabled = True
-                pa_widgets.scalar_cmap.disabled = True
-                pa_widgets.velocity_vectors.disabled = True
-                pa_widgets.vector_size.disabled = True
-                pa_widgets.scalar_size.disabled = True
+                pa_widgets.scalar.disabled = False
+                pa_widgets.scalar_cmap.disabled = False
+                pa_widgets.velocity_vectors.disabled = False
+                pa_widgets.vector_size.disabled = False
+                pa_widgets.scalar_size.disabled = False
+                pa_widgets.is_visible.disabled = False
+
+    def _normalized_for_colormaps(self, data):
```

Comment:
This is the same function as before, Don't Repeat Yourself (DRY).  Please refactor this to be reused.

Reply:
I've added this function to the base class `Viewer`, from which both the `2DViewer` and the `3DViewer` classes inherit. This takes care of this issue.